communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java

```diff
@@ -39,13 +49,43 @@
 @ApplicationScoped
 public class AproxConnectorImpl implements AproxConnector {
 
-    private final Configuration config = new Configuration();
+    @Inject
+    private Configuration config;
+
+    @Inject
+    private SCM scmManager;
+
+    @Inject
+    private PomAnalyzer pomAnalyzer;
 
     @Override
-    public Optional<GAVDependencyTree> getDependencyTreeOfRevision(String scmUrl, String revision,
-            String pomPath) {
-        // TODO Auto-generated method stub
-        return Optional.empty();
+    public Optional<List<ScmGAVRelationship>> getDependencyTreeOfRevision(String scmUrl,
+            String revision, String pomPath) throws ScmException, PomAnalysisException {
+
+        try {
+            // git clone
+            // TODO: hardcoded to git right now
+            File tempDir = Files.createTempDirectory("cloned_repo").toFile();
```

Comment:
It would be great to make it safe to run DA more times on the same VM (that's our real deployment - two EAPs on th same VM). I think it is not safe right now, or am I wrong?